Classes/GarbageCollector.php

```diff
@@ -155,6 +155,20 @@ public function processDatamap_afterDatabaseOperations($status, $table, $uid, ar
 			|| ($table == 'pages' && !$this->isIndexablePageType($record))
 		) {
 			$this->collectGarbage($table, $uid);
+
+			// if the start time is in the future and not otherwise invisible,
+			// it is necessary to add the item to the queue again to be indexed once the start time is reached
+			if ($this->isStartTimeInFuture($table, $record)
```

Comment:
Ah, now I understand you where the first isHidden check is done.

It might make sense to put it before?
Because e.g. if the record is also set to "hidden" we have to delete it from the queue.

However, that means it could be that the following happens:
1.) delete item
2.) remove from queue due to future starttime
3.) re-add item to queue due to future start time
4.) due to hidden remove it from queue again.

But I think that would be acceptable?
In which case it would only be the future und isindexed check and maybe the tables check.